TreeDB/collections/vector_index_persist.go

```
@@ -0,0 +1,1609 @@
+			}
+			return status, true, nil
+		}
+		return status, false, nil
+	}
+	if idx.isNativePersistent() || collectionMetaDeclaresVectorIndex(c.meta, idx.name) {
+		status.ExactFallbackReason = vectorIndexFallbackStaleRuntimeIndex
+		return status, true, nil
+	}
+	declared, err := c.refreshVectorIndexDeclaration(idx.name)
+	if err != nil || !declared {
+		return status, false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 127) but it returns nil

(nilerr)